crates/llms/src/anthropic/mod.rs

```diff
@@ -0,0 +1,184 @@
+*/
+
+use async_openai::{config::Config, Client};
+use reqwest::header::{HeaderMap, HeaderValue, AUTHORIZATION, CONTENT_TYPE};
+use secrecy::{ExposeSecret, Secret};
+use std::sync::LazyLock;
+
+mod chat;
+mod types;
+mod types_stream;
+
+pub use types::AnthropicModelVariant;
+
+pub struct Anthropic {
+    client: Client<AnthropicConfig>,
+    model: AnthropicModelVariant,
+
+    // The name of the model as known in the spice runtime (not the anthropic model).
+    name: String,
+}
+
+static ANTHROPIC_API_BASE: &str = "https://api.anthropic.com/v1";
+pub static DEFAULT_ANTHROPIC_MODEL: &str = "claude-3-5-sonnet-20240620";
+static ANTHROPIC_API_VERSION: &str = "2023-06-01";
+static DUMMY_API_KEY: LazyLock<Secret<String>> = LazyLock::new(|| Secret::new(String::new()));
+
+impl Anthropic {
+    pub fn new<S: Into<AnthropicModelVariant>>(
+        config: AnthropicConfig,
+        model: S,
+        name: &str,
+    ) -> Self {
+        Self {
+            client: Client::<AnthropicConfig>::with_config(config),
+            model: model.into(),
+            name: name.to_string(),
+        }
+    }
+}
+
+#[derive(Clone)]
+pub struct AnthropicConfig {
+    pub auth: Option<AnthropicAuthMechanism>,
+    pub base_url: String,
+    pub version: String,
+    pub beta: Option<Vec<String>>,
+}
+
+impl Default for AnthropicConfig {
+    fn default() -> Self {
+        Self {
+            auth: None,
+            base_url: ANTHROPIC_API_BASE.to_string(),
+            version: ANTHROPIC_API_VERSION.to_string(),
+            beta: None,
+        }
+    }
+}
+
+impl AnthropicConfig {
+    #[must_use]
+    pub fn new() -> Self {
+        AnthropicConfig::default()
+    }
+
+    #[must_use]
+    pub fn with_api_key<S: Into<String>>(mut self, api_key: Option<S>) -> Self {
+        if let Some(api_key) = api_key {
+            self.auth = Some(AnthropicAuthMechanism::ApiKey(Secret::new(api_key.into())));
+        }
+        self
+    }
+
+    #[must_use]
+    pub fn with_auth_token<S: Into<String>>(mut self, auth_token: Option<S>) -> Self {
+        if let Some(auth_token) = auth_token {
+            self.auth = Some(AnthropicAuthMechanism::AuthToken(Secret::new(
+                auth_token.into(),
+            )));
+        }
+        self
+    }
+
+    #[must_use]
+    pub fn with_base_url<S: Into<String>>(mut self, base_url: Option<S>) -> Self {
+        if let Some(base_url) = base_url {
+            self.base_url = base_url.into();
+        }
+        self
+    }
+
+    #[must_use]
+    pub fn with_version<S: Into<String>>(mut self, version: Option<S>) -> Self {
+        if let Some(version) = version {
+            self.version = version.into();
+        }
+        self
+    }
+
+    #[must_use]
+    pub fn with_beta(mut self, beta: Vec<String>) -> Self {
+        self.beta = Some(beta);
+        self
+    }
+}
+
+#[derive(Clone)]
+pub enum AnthropicAuthMechanism {
+    ApiKey(Secret<String>),
+    AuthToken(Secret<String>),
+}
+
+#[allow(clippy::expect_used)]
```

Comment:
```suggestion
```

Reply:
Right now this has to accept having an `.expect()` because it need to match the `Config` trait
```rust
fn headers(&self) -> HeaderMap;
```

Reply:
But i will leave a note

Reply:
You can use an explicit `panic!` for this case